### PR TITLE
gh-105323: Remove `WITH_APPLE_EDITLINE` to use the same declaration for all editline

### PR DESCRIPTION
**Pull request by gaogaotiantian**
See the discussion in https://github.com/python/cpython/issues/105323#issuecomment-1831052024

* Issue: gh-105323

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2023-11-29-04-43-32.gh-issue-105323.zelrW9.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove ``WITH_APPLE_EDITLINE`` to use the same declaration for all ``editline``
```
Comment (corona10) [inline Misc/NEWS.d/next/Library/2023-11-29-04-43-32.gh-issue-105323.zelrW9.rst]: No end user impact. We can skip NEWS.d

Comment (gaogaotiantian) [reply Misc/NEWS.d/next/Library/2023-11-29-04-43-32.gh-issue-105323.zelrW9.rst]: Sure, do you want me to remove this file and label the PR?

Comment (corona10) [reply Misc/NEWS.d/next/Library/2023-11-29-04-43-32.gh-issue-105323.zelrW9.rst]: > Sure, do you want me to remove this file and label the PR?

I already added the label, just please remove the file :)

Comment (gaogaotiantian) [reply Misc/NEWS.d/next/Library/2023-11-29-04-43-32.gh-issue-105323.zelrW9.rst]: File removed :)

**Comment by corona10** [comment]
I will think about little bit more about side effect from this change :)
Please give me a time for this weekend

**Comment by gaogaotiantian** [comment]
No problem.

The readline/editline stuff in `readline` is a bit twisted. I believe at least as of now we never use the real `editline.h`. We use either `readline/readline.h`, which could be either from the real GNU readline or the "emulated" Mac libedit, or `editline/readline.h`, which is the compatible header from `editline` - it could be `libedit-dev` or some libedit om Mac.

`WITH_EDITLINE` does not make the situation clearer.

**Review by erlend-aasland — APPROVED** [review]

**Review by corona10 — APPROVED** [review]

**Pull request merged**